### PR TITLE
still trying to get patch and create to pass

### PR DESCRIPTION
**Pull request by AbdullahAlger**
Still trying to get the tests for patch and create to pass. Having some issues there with the parameters.

### CODE REVIEW

File: spec/controllers/wikis_controller_spec.rb [hunk 1]
```diff
@@ -56,13 +58,24 @@
     xit "updates a wiki" do
       user = create(:user)
       sign_in(user)
+      wiki = create(:wiki, user: user)
+
+      patch :update, id: wiki
```
Comment (eliotsykes) [inline spec/controllers/wikis_controller_spec.rb]: Due to [this line in `WikisController#update`](https://github.com/AbdullahAlger/blocipedia/blob/rspec-controller-tests/app/controllers/wikis_controller.rb#L39), try this

```ruby
patch :update, id: wiki.friendly_id
```

File: spec/controllers/wikis_controller_spec.rb [hunk 2]
```diff
@@ -41,14 +46,11 @@
     xit "creates a wiki" do
       user = create(:user)
       sign_in(user)
+      wiki = create(:wiki, user: user)
```
Comment (eliotsykes) [inline spec/controllers/wikis_controller_spec.rb]: Line 49 can be removed, the controller will create the wiki.

**Comment by AbdullahAlger** [comment]
I think there is something wrong because I've tried many things. I have tried your suggestion as well, but `friendly_id` is not in my DB, but `slug` is, which is what `friendly_id` uses. I am getting caught up with `post :create`, but I will try working on patch now.

File: spec/controllers/wikis_controller_spec.rb [hunk 3]
```diff
@@ -56,13 +58,24 @@
     xit "updates a wiki" do
       user = create(:user)
       sign_in(user)
+      wiki = create(:wiki, user: user)
+
+      patch :update, id: wiki
+      expect(response).to have_http_status(:found)
+      expect(assigns(:wiki)).to eq(wiki)
```
Comment (eliotsykes) [inline spec/controllers/wikis_controller_spec.rb]: The `wiki` won't be available as `assigns(:wiki)` as its a `local_variable` in `WikisController#update`. Only `@instance_variables` are accessible through `assigns(...)` in a test.

Comment (AbdullahAlger) [inline spec/controllers/wikis_controller_spec.rb]: @eliotsykes 
Might want to check out my recent commit on this. I got it to work, but I am not sure why it works. It's probably wrong and I've found a workaround.

**Pull request merged**